src/pointer/ptr.rs

```diff
@@ -855,45 +869,45 @@ mod _casts {
         }
     }
 
-    impl<'a, T, I, const N: usize> Ptr<'a, [T; N], I>
+    impl<'a, V, T, I, const N: usize> Ptr<'a, V, I>
     where
-        T: 'a,
+        V: Validity<Inner = [T; N]>,
         I: Invariants,
     {
         /// Casts this pointer-to-array into a slice.
         #[allow(clippy::wrong_self_convention)]
-        pub(crate) fn as_slice(self) -> Ptr<'a, [T], I> {
+        pub(crate) fn as_slice(self) -> Ptr<'a, V::WithInner<[T]>, I> {
             let slice = self.as_inner().as_slice();
             // SAFETY: Note that, by post-condition on `PtrInner::as_slice`,
             // `slice` refers to the same byte range as `self.as_inner()`.
             //
-            // 6. Thus, `slice` conforms to the aliasing invariant of
+            // 6. By the above lemma, `slice` conforms to the validity invariant
+            //    of `V` because `self` does.
+            // 7. Thus, `slice` conforms to the aliasing invariant of
             //    `I::Aliasing` because `self` does.
-            // 7. By the above lemma, `slice` conforms to the alignment
+            // 8. By the above lemma, `slice` conforms to the alignment
             //    invariant of `I::Alignment` because `self` does.
-            // 8. By the above lemma, `slice` conforms to the validity invariant
-            //    of `I::Validity` because `self` does.
             unsafe { Ptr::from_inner(slice) }
         }
     }
 
     /// For caller convenience, these methods are generic over alignment
     /// invariant. In practice, the referent is always well-aligned, because the
     /// alignment of `[u8]` is 1.
-    impl<'a, I> Ptr<'a, [u8], I>
+    impl<'a, I> Ptr<'a, Valid<[u8]>, I>
     where
-        I: Invariants<Validity = Valid>,
+        I: Invariants,
     {
-        /// Attempts to cast `self` to a `U` using the given cast type.
+        /// Attempts to cast `self` to a `W` using the given cast type.
```

Comment:
Clarify that it's an `Initialized<W>`, and probably say that it's into a `Ptr<Initialized<W>>`.

Reply:
Done.